prac_07/project_management.py

```diff
@@ -0,0 +1,233 @@
+"""
+prac_07 - project_management.py
+
+Estimated time: 120 mins
+Actual time: 118 mins
+"""
+
+import datetime
+from prac_07.project import Project
+
+MENU = ("- (L)oad projects\n- (S)ave projects\n- (D)isplay projects\n- (F)ilter projects by date\n"
+        "- (A)dd new project\n- (U)pdate project\n- (Q)uit")
+DEFAULT_FILENAME = "projects.txt"
+
+
+def main():
+    """Run a program that load and save a data file and use a list of Project objects."""
+    projects = load_projects(DEFAULT_FILENAME)
+    print(MENU)
+
+    choice = input(">>> ").upper()
+    while choice != "Q":
+        if choice == "L":
+            filename = get_valid_filename()
+            new_projects = load_projects(filename)
+            for project in new_projects:
```

Comment:
Should we have this low level detail in `main()`? Could we refactor it into the loading function or into another function?

Reply:
I did not put get_valid_file_name in load_projects so that I can reuse it at the start of the program where it loads the projects from a default file. Unless there was another way?

Reply:
The way I implemented it was with the filename argument of `load_projects` having a default of `""`, so if no parameter is given, it will get the filename from the user, with a `if filename == ""`. That way it can take in the filename for loading the initial projects and also get a filename from the user if none is specified. I hope that made sense.

Reply:
Oh yes! I completely forgot about that 🙈. Thank you!

Reply:
Actually, If I do what you suggest, it will breach my function SRP. 
In your approach, you are getting a filename **and** loading the projects?
@alexchristie12